doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/examples/ml-example.md

```diff
@@ -1,10 +1,253 @@
 (kuberay-ml-example)=
 
-# Example machine learning workloads
+# XGBoost-Ray on Kubernetes
 
-:::{warning}
-This page is under construction!
+:::{note}
+To learn the basics of Ray on Kubernetes, we recommend taking a look
+at the {ref}`introductory guide<kuberay-quickstart>` first.
 :::
 
-At least one end-to-end example of an actual machine learning workload,
-preferably with GPUs, possibly engaging the autoscaling functionality.
+In this guide, we show you how to run a sample Ray machine learning
+workload on Kubernetes infrastructure.
+
+We will run Ray's {ref}`XGBoost training benchmark<xgboost-benchmark>` with a 100 gigabyte training set.
+* Learn more about {ref}`XGBoost-Ray<xgboost-ray>`.
+
+## A note on autoscaling
+This guide will show you how to run an XGBoost workload with and without optional Ray Autoscaler support.
+When choosing whether or not to use autoscaling functionality, keep the following considerations in mind.
+
+### Autoscaling: Pros
+
+#### Cope with unknown resource requirements.
+If you don't know how much compute your Ray workload will require,
+autoscaling can adjust your Ray cluster to the right size.
+
+#### Save on costs.
+Idle compute is automatically scaled down, potentially leading to cost savings.
+
+### Autoscaling: Cons
+
+#### Less predictable when resource requirements are known.
+If you already know exactly how much compute your workload requires, it makes
+sense to provision a statically-sized Ray cluster.
+In this guide's example, we know that we need 1 Ray head and 9 Ray workers,
+so autoscaling is not strictly required.
+
+#### Longer end-to-end runtime.
+Autoscaling entails provisioning compute for Ray workers while the Ray application
+is running. On the other hand, if you pre-provision a fixed number of Ray nodes,
+all of the Ray nodes can be started in parallel, potentially reducing your application's
+runtime.
+
+## Kubernetes infrastructure setup
+
+### Managed Kubernetes services
+
+Running the example in this guide requires some basic Kubernetes infrastructure set-up.
+We collect some helpful links for users who are getting started with a managed Kubernetes service.
+
+#### Landing pages
+Landing pages for the major cloud providers' Kubernetes services can be found at the following links.
+- [AKS landing page (Azure)](https://azure.microsoft.com/en-us/services/kubernetes-service/)
+- [EKS landing page (Amazon Web Services)](https://aws.amazon.com/eks/)
```

Comment:
We should probably put EKS and GKE before AKS here otherwise people might think AKS is better supported than the others (which is certainly not the case) :D

Reply:
I was shooting for alphabetical order.
My order of familiarity is GKE, EKS, AKS.

Reply:
That might be the historical order of product launch.